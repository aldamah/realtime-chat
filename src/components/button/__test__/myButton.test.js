import React from "react";
import ReactDOM  from "react-dom";
import MyButton from "../myButton";
import { render, cleanup } from "@testing-library/react";

import renderer from "react-test-renderer";

afterEach(cleanup);

it(" renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<MyButton/>, div);
});

it("renders  button correctly", () => {
    const { getByTestId } = render(<MyButton label={"Click here"} />);
    expect(getByTestId("mybutton")).toHaveTextContent("Click here");
});

it ("matches snapshot", () => {
   const tree = renderer.create(<MyButton label={"save"}/>).toJSON();
   expect(tree).toMatchSnapshot();
});