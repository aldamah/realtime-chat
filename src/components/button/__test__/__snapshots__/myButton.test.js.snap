// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="mybutton-style"
  data-testid="mybutton"
>
  save
</div>
`;
